enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,361 @@
+config secret while also respecting the environment variables that would be set by the webhook. Additionally, upstream projects may reject the notion of reading these details from a config secret but that has yet to be seen.
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+An e2e test job will be created similar to the [e2e-gcp-manual-oidc](https://github.com/openshift/release/pull/22552) that,
+- Extracts `CredentialsRequests` from the release image.
+- Processes `CredentialsRequests` with `ccoctl` to generate secret and `Authentication` configuration manifests.
+- Moves the generated manifests into the manifests directory used for install.
+- Runs the normal e2e suite against the resultant cluster.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+Azure workload identity will be introduced as [TechPreviewNoUpgrade](https://github.com/openshift/api/blob/fefb3487546079495fb80ca0f1155ecd7417b9d8/config/v1/types_feature.go#L111) and then promoted once it is demonstrated to be working reliably.
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+As clusters are upgraded, new permissions may be required or extended (in the case of future role granularity work) and users must evaluate those changes at the upgrade boundary similarly to [upgrading an STS cluster in manual mode](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html#manual-mode-sts-blurb).
+
+### Version Skew Strategy
+
+None.
+
+### Operational Aspects of API Extensions
+
+None.
+
+#### Failure Modes
+
+None.
+
+#### Support Procedures
+
+##### How to detect that operator credentials are incorrect / insufficient?
+
+Operators will be degraded when credentials are insufficient / incorrect because operators will be unable to authenticate using the provided credentials or the permissions granted to the associated identity were insufficient. CCO will not monitor the state of the credentials on-cluster because CCO will be disabled based on clusters operating in `manual` credentials mode.
```

Comment:
https://issues.redhat.com/browse/CCO-370
```
As an openshift developer I need to know that my operator goes degraded if the Azure Managed Identity is insufficient or incorrect so that my operator is admin friendly in clusters that leverage Azure Managed Identity.

Acceptance criteria:

Serial test that picks a random operator of those identified and updates it to use an invalid role reference, verify the operator goes Degraded in a timely manner
Serial test that picks a random operator of those identified and updates it to use a role with insufficient permissions (null or otherwise)
```

@deads2k sound good?